bpo-23677: Mention dict and set comprehension in stdtypes (#20027)
python/cpython: Doc/library/stdtypes.rst

a dictionary key or as an element of another set.

Non-empty sets (not frozensets) can be created by placing a comma-separated list
of elements within braces, for example: ``{'jack', 'sjoerd'}``, by using a set

> comprehension, for example ``{x for x in iterable}``, in addition to the

Reviewer: This should be two separate sentences.  Also, set comprehensions can make both empty and non-empty sets.

Author: Thank you for your review. Can you review the changes again?

a dictionary key or as an element of another set.

Non-empty sets (not frozensets) can be created by placing a comma-separated list

> of elements within braces, for example: ``{'jack', 'sjoerd'}``.  Non-empty sets can

Reviewer: Non-empty sets -> Sets

```
>>> {x for x in range(0)}
set()
```

Author: I have made the requested changes; please review again.

> be created by using a set comprehension, for example ``{x for x in iterable}``.

Reviewer 2: ```suggestion
be created by using a set comprehension, for example ``{2 * x for x in iterable}``.
```

Otherwise you can just do set(iterable).

> be created by using a setcomprehension, for example ``{x for x in iterable}``.

Reviewer 2: setcomprehension --> set comprehension

> of elements within braces, for example: ``{'jack', 'sjoerd'}``.  Sets can
> be created by using a set comprehension, for example ``{2 * x for x in iterable}``.
> Also, set comprehensions can make both empty and non-empty sets.  In addition to
> the :class:`set` constructor.

Reviewer 3: ```suggestion
of elements within braces, for example ``{'jack', 'sjoerd'}``.  Sets can
be created by using a :term:`set comprehension`, for example ``{2 * x for x in iterable}``,
or using the :class:`set` constructor.
```
- Remove unnecessary sentence about creating empty sets using a setcomp.
- Fix incomplete sentence
- Link to the glossary

> the :class:`set` constructor.

Reviewer 2: There is no need to say "Non-empty sets (not frozensets) can be created by using a set comprehension..." and then add "Also, set comprehensions can make both empty and non-empty sets". Just remove the "non-empty from the first sentence and delete the whole second sentence.

Dictionaries can be created by placing a comma-separated list of ``key: value``
pairs within braces, for example: ``{'jack': 4098, 'sjoerd': 4127}`` or ``{4098:
'jack', 4127: 'sjoerd'}``, by a dict comprehension, for example:

> ``{x: x**2 for x in iterable}`` or by the :class:`dict` constructor.

Reviewer: Split into three sentences,and add an example for the dict constructor:  ``dict(jack=4098, sjoerd= 4127}``

Dictionaries can be created by placing a comma-separated list of ``key: value``
pairs within braces, for example: ``{'jack': 4098, 'sjoerd': 4127}`` or ``{4098:
'jack', 4127: 'sjoerd'}``, by a dict comprehension, for example: ``{x: x**2 for

> x in iterable}``.  Dictionaries can be created by the :class:`dict` constructor,

Reviewer 2: ```suggestion
x in iterable}``.  They can also be created by the :class:`dict` constructor,
```

Reviewer: The "also" is fine, but I prefer the precise "Dictionaries" over "they".

Author: I think "Dictionaries" are more sense.  @rhettinger I made changes. Can you review it again?

> 'jack', 4127: 'sjoerd'}``, by a dict comprehension, for example: ``{x: x**2 for
> x in iterable}``.  Dictionaries can also be created by the :class:`dict`
> constructor, for example: ``dict(jack=4098, sjoerd=4127)``.

Reviewer 3: ```suggestion
'jack', 4127: 'sjoerd'}``, or by a :term:`dictionary comprehension`, for example: ``{x: x**2 for
x in iterable}``.  Dictionaries can also be created by the :class:`dict`
constructor, for example: ``dict(jack=4098, sjoerd=4127)``.
```
Link the glossary term
